shallow_backup.py

```diff
@@ -179,27 +186,69 @@ def backup_dotfiles(backup_path):
 
 	# Sublime Text Configs
 	if os.path.isdir("/Users/alichtman/Library/Application Support/Sublime\ Text\ 2"):
-		dotfolders_mp_in.append((os.path.join(home_path, "Library/Application\ Support/Sublime\ Text\ 2/Packages/User"), backup_path))
+		dotfolders_mp_in.append(
+			(os.path.join(home_path, "Library/Application\ Support/Sublime\ Text\ 2/Packages/User"), backup_path))
 
 	if os.path.isdir("/Users/alichtman/Library/Application Support/Sublime\ Text\ 3"):
-		dotfolders_mp_in.append((os.path.join(home_path, "Library/Application\ Support/Sublime\ Text\ 3/Packages/User"), backup_path))
+		dotfolders_mp_in.append(
+			(os.path.join(home_path, "Library/Application\ Support/Sublime\ Text\ 3/Packages/User"), backup_path))
 
 	# pprint(dotfiles_mp_in)
 	# pprint(dotfolders_mp_in)
 
 	# Multiprocessing
 	with mp.Pool(mp.cpu_count()):
 
-		print(Fore.BLUE + Style.BRIGHT + "Backing up dotfolders..." + Style.RESET_ALL)
+		print(Fore.BLUE + Style.BRIGHT +
+			  "Backing up dotfolders..." + Style.RESET_ALL)
 		for x in dotfolders_mp_in:
 			x = list(x)
 			mp.Process(target=copy_dotfolder, args=(x[0], x[1],)).start()
 
 	with mp.Pool(mp.cpu_count()):
-		print(Fore.BLUE + Style.BRIGHT + "Backing up dotfiles..." + Style.RESET_ALL)
+		print(Fore.BLUE + Style.BRIGHT +
+			  "Backing up dotfiles..." + Style.RESET_ALL)
 		for x in dotfiles_mp_in:
 			x = list(x)
-			mp.Process(target=copy_dotfile, args=(x[0], x[1],)).start()
+			mp.Process(target=_copy_file, args=(x[0], x[1],)).start()
+
+
+def backup_configs(backup_path):
+	make_dir_warn_overwrite(backup_path)
+
+	def home_prefix(path):
+		return os.path.join(os.path.expanduser('~'), path)
+
+	configs_dir_mapping = {"Library/Application Support/Sublime Text 2/Packages/User/": "sublime_2",
+						   "Library/Application Support/Sublime Text 3/Packages/User/": "sublime_3", }
+	plist_files = ["Library/Preferences/com.apple.Terminal.plist"]
+
+	# backup config dirs in backup_path/configs/<target>/
+	for config, target in configs_dir_mapping.items():
+		if os.path.isdir(home_prefix(config)):
+			configs_backup_path = os.path.join(backup_path, target)
+			_mkdir_or_pass(configs_backup_path)
+			_copy_dir_content(home_prefix(config), configs_backup_path)
+
+	# backup plist files in backup_path/configs/plist/
+	plist_backup_path = os.path.join(backup_path, "plist")
+	_mkdir_or_pass(plist_backup_path)
+	for plist in plist_files:
+		if os.path.exists(home_prefix(plist)):
+			_copy_dir_content(home_prefix(plist), plist_backup_path)
+
+
+def _copy_dir_content(source, target):
+	"""Copies the contents of a dir to a specified target path."""
+	cmd = "cp -a '" + source + "' '" + target + "/'"
+	# print(cmd)
+	sp.run(cmd, shell=True, stdout=sp.PIPE)
+
+
+def _mkdir_or_pass(dir):
```

Comment:
This function should be moved to near the top of the file by the other utility functions